R/cluster.R

```diff
@@ -148,29 +146,42 @@ makeCluster <- function(clusterSetting = "cluster_settings.json", fullName = FAL
     packages <- c(installCranCommand, installGithubCommand)
   }
 
+  if (!is.null(poolConfig[["pool"]])) {
+    validateDeprecatedClusterConfig(clusterSetting)
+    poolConfig <- poolConfig[["pool"]]
+  }
+  else {
+    validateClusterConfig(clusterSetting)
+  }
+
   response <- .addPool(
-    pool = pool,
+    pool = poolConfig,
     packages = packages,
     resourceFiles = resourceFiles)
 
-  pool <- getPool(pool$name)
+  pool <- rAzureBatch::getPool(poolConfig$name)
 
   if (grepl("AuthenticationFailed", response)) {
     stop("Check your credentials and try again.");
   }
 
   if (grepl("PoolExists", response)) {
-    print("The specified pool already exists. Will use existing pool.")
+    cat("The specified pool already exists. Will use existing pool.", fill = TRUE)
+
+    if (pool$targetDedicatedNodes !=  poolConfig$poolSize$dedicatedNodes || pool$targetLowPriorityNodes != poolConfig$poolSize$lowPriorityNodes) {
+      warning("There is a mismatched between cluster config nodes and existing pool target nodes")
```

Comment:
Give more info as to what the mismatch is. This doesn't give the user enough info to address the issue.